src/azure_devtools/scenario_tests/preparers.py

```diff
@@ -46,11 +49,15 @@ def _preparer_wrapper(test_class_instance, **kwargs):
 
             trim_kwargs_from_test_function(fn, kwargs)
 
+            error = False
             try:
                 fn(test_class_instance, **kwargs)
+            except Exception:  # pylint: disable=broad-except
+                error = True
             finally:
                 # Russian Doll - the last declared resource to be deleted first.
-                self.remove_resource_with_record_override(resource_name, **kwargs)
+                if not (self.recover and error):
+                    self.remove_resource_with_record_override(resource_name, **kwargs)
```

Comment:
Should there be a warning message that the resource will not be cleaned up if self.recover and error is True as we may forget to turn it off?

Reply:
Added